knowledge-base/architectural-decisions-records/sdk-2.0.md

```diff
@@ -0,0 +1,64 @@
+# Action plan for Version 2.0 of the SAP Cloud SDK
+
+We want to prepare the release of a major version update. The following points need to be considered for this:
+
+## Remove all deprecated functionality from public api (#after)
+* Make sure to not remove deprecated functionality where the api should only be hidden
+
+## Export only api relevant functionality to the root (#after, prep steps before)
+* Find a way to make clear what is public facing api and what is not (#before)
+* Investigate what should be private and deprecate it as soon as possible (#before)
+
+# Replace connectivity with functionality from @sap/xssec (or other SAP libraries) (#before)
+* And deprecate old functionality
+
+## Separate core into multiple npm packages (#after, prep steps before)
+ * `@sap-cloud-sdk/odata-common`
+ * `@sap-cloud-sdk/odata-v2`
+ * `@sap-cloud-sdk/odata-v4`
+ * `@sap-cloud-sdk/connectivity`
+ * `@sap-cloud-sdk/http-client`
+If needed maybe also `@sap-cloud-sdk/core` but hopefully not needed.
+* Sort packages into separate directories beforehand (#before)
+  * Mimic packages (without package.json and co) in the core directory
+  * Circular depencies - make sure that there are only imports from the whole module, e. g.
+    In odata-v2 package, if I want to import X from http-client import it from `http-client` not `http-client/X`
+* The connectivity package currently has only the cf scope. If at some point we add other systems (e. g. kyma) this will be added here.
+
+### Package dependencies (#after, conceptually #before)
+The packages above should rely on one another like this (dependant -> dependency):
+* `@sap-cloud-sdk/odata-v2` -> `@sap-cloud-sdk/odata-common`
+* `@sap-cloud-sdk/odata-v4` -> `@sap-cloud-sdk/odata-common`
+* `@sap-cloud-sdk/odata-common` -> `@sap-cloud-sdk/connectivity`
+* `@sap-cloud-sdk/odata-common` -> `@sap-cloud-sdk/http-client`
+* `@sap-cloud-sdk/http-client` -> `@sap-cloud-sdk/connectivity`
+
+### Other related activities (#after)
+Rename all functionality with V2/V4 suffix, e. g.:
```

Comment:
maybe update a bit, to reflect the latest plan:
- rename now, but export `GetAllRequestBuilderV2`.
- adjust the export later